### ISSUE TITLE
Icons from Slack snap app not rendering

### ISSUE THREAD

**Issue by owais** [Ubuntu/gnome-shell-extension-appindicator#115]
I installed the recently published snap from slack. Everything works fine except the icon in the panel is blank. The indicator is there and it works but the icon is invisible as if it was transparent. I tried running the same snap under budgie desktop and it worked fine there.

![screenshot from 2018-01-19 02-44-22](https://user-images.githubusercontent.com/46186/35199396-fa0fcfde-ff22-11e7-901a-3d91ffd6cc54.png)

**Comment by simonvanderveldt** [comment]
Sorry to kind of hijack this, but did you have to do anything to get Slack to show up at all?
I'm not seeing Slack at all (with or without icon), doesn't matter if I set `XDG_CURRENT_DESKTOP=Unity` or not.

**Comment by owais** [comment]
I can get the regular (DEB) slack app to show up by changing Exec line to the following in /usr/share/applications/slack.desktop

`Exec=env XDG_CURRENT_DESKTOP=Unity /usr/bin/slack --disable-gpu %U`

**Comment by simonvanderveldt** [comment]
I figured it out, the Slack binary is actually still using gtk2 :man_facepalming: which was disabled for my libindicator/libappindicator builds. Sorry for the noise.

Regarding your issue, I image it might be caused by the `XDG_CURRENT_DESKTOP` env var not getting to the application. Electron should really just check if the `org.kde.StatusNotifierWatcher` D-Bus interface exists though...

**Comment by balazsorban44** [comment]
I have the same issue. :cry:  @simonvanderveldt I am a bit confused about your message, I am a noob.. But did you manage to solve the problem, or not? Or now it shows up for you as well, just only with a transparent icon?

**Comment by simonvanderveldt** [comment]
@balazsorban44 No, it's working for me. I was missing the gtk2 version of libappindicator.
It's been fixed now in my distro (gentoo) so if you run gentoo as well you should be fine, otherwise make sure you have the gt2 version of libappindicator installed.

**Comment by balazsorban44** [comment]
Could you help me out how to install it on Ubuntu? :crossed_fingers:

**Comment by simonvanderveldt** [comment]
@balazsorban44 No, sorry, I have no experience with Ubuntu.

**Comment by owais** [comment]
@simonvanderveldt Are you using Slack from the snap store? Slack does show icon on Ubuntu when installed directly from the DEB. This issue is related to the snap store version.

**Comment by gokuatkai** [comment]
Using Ubuntu 18.04.2 LTS with xfce (xubuntu), installed from .deb and it does not show icon either. 
v3.4.2 64b
